engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableFloatingPointZeroTest.java

```diff
@@ -0,0 +1,599 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static io.deephaven.engine.testutil.TstUtils.assertTableEquals;
+import static io.deephaven.engine.util.TableTools.doubleCol;
+import static io.deephaven.engine.util.TableTools.floatCol;
+import static io.deephaven.engine.util.TableTools.intCol;
+import static io.deephaven.engine.util.TableTools.longCol;
+import static io.deephaven.engine.util.TableTools.newTable;
+
+@Category(OutOfBandTest.class)
+public class QueryTableFloatingPointZeroTest {
+
+    @Rule
+    public final EngineCleanup base = new EngineCleanup();
+
+    @Before
+    public void setUp() throws Exception {
+        ChunkPoolReleaseTracking.enableStrict();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        ChunkPoolReleaseTracking.checkAndDisable();
+    }
+
+    @Test
+    public void testFloatJoinKey() {
+        final Table t1 = newTable(floatCol("KeyNeg", -0.0f), floatCol("Neg", -0.0f));
+        final Table t2 = newTable(floatCol("KeyPos", 0.0f), floatCol("Pos", 0.0f));
+        for (Table out : Arrays.asList(
+                t1.naturalJoin(t2, "KeyNeg=KeyPos"),
+                t2.naturalJoin(t1, "KeyPos=KeyNeg"),
+                t1.exactJoin(t2, "KeyNeg=KeyPos"),
+                t2.exactJoin(t1, "KeyPos=KeyNeg"),
+                t1.join(t2, "KeyNeg=KeyPos"),
+                t2.join(t1, "KeyPos=KeyNeg"),
+                t1.aj(t2, "KeyNeg>=KeyPos"),
+                t2.aj(t1, "KeyPos>=KeyNeg"))) {
+            final long key = oneKey(out);
+            floatToBitsEquals(out, "KeyNeg", key, -0.0f);
+            floatToBitsEquals(out, "KeyPos", key, 0.0f);
+            floatToBitsEquals(out, "Neg", key, -0.0f);
+            floatToBitsEquals(out, "Pos", key, 0.0f);
+        }
+    }
+
+    @Test
+    public void testDoubleJoinKey() {
+        final Table t1 = newTable(doubleCol("KeyNeg", -0.0), doubleCol("Neg", -0.0));
+        final Table t2 = newTable(doubleCol("KeyPos", 0.0), doubleCol("Pos", 0.0));
+        for (Table out : Arrays.asList(
+                t1.naturalJoin(t2, "KeyNeg=KeyPos"),
+                t2.naturalJoin(t1, "KeyPos=KeyNeg"),
+                t1.exactJoin(t2, "KeyNeg=KeyPos"),
+                t2.exactJoin(t1, "KeyPos=KeyNeg"),
+                t1.join(t2, "KeyNeg=KeyPos"),
+                t2.join(t1, "KeyPos=KeyNeg"),
+                t1.aj(t2, "KeyNeg>=KeyPos"),
+                t2.aj(t1, "KeyPos>=KeyNeg"))) {
+            final long key = oneKey(out);
+            doubleToBitsEquals(out, "KeyNeg", key, -0.0);
+            doubleToBitsEquals(out, "KeyPos", key, 0.0);
+            doubleToBitsEquals(out, "Neg", key, -0.0);
+            doubleToBitsEquals(out, "Pos", key, 0.0);
+        }
+    }
+
+    @Test
+    public void testFloatJoinMultiKey() {
+        final Table t1 = newTable(intCol("Key1", 0), floatCol("KeyNeg", -0.0f), floatCol("Neg", -0.0f));
```

Comment:
I think you may want two fp keys. Mix and match double and float in both orders?

Reply:
hoping a single float key in a variety of positions is good enough.